core/src/components/select/select.tsx

```diff
@@ -1138,9 +921,86 @@ Developers can use the "legacy" property to continue using the legacy form marku
   }
 
   render() {
```

Comment:
I didn't change the `renderSelect()` function but I moved it down in the file to where the `render()` function was which is why there is a larger diff